client/my-sites/hosting/main.js

```diff
@@ -184,207 +183,174 @@ const SidebarCards = ( { isBasicHostingDisabled } ) => {
 	return <ShowEnabledFeatureCards cards={ sidebarCards } availableTypes={ availableTypes } />;
 };
 
-class Hosting extends Component {
-	state = {
-		clickOutside: false,
-	};
-
-	handleClickOutside( event ) {
-		const { COMPLETE } = transferStates;
-		const { isTransferring, transferState } = this.props;
-
-		if ( isTransferring && COMPLETE !== transferState ) {
-			event.preventDefault();
-			event.stopImmediatePropagation();
-			this.setState( { clickOutside: true } );
-		}
-	}
-
-	componentDidMount() {
-		const { COMPLETE } = transferStates;
-		// Check if a reverted site still has the COMPLETE status
-		if ( this.props.transferState === COMPLETE ) {
-			// Try to refresh the transfer state
-			this.props.fetchAutomatedTransferStatus( this.props.siteId );
-		}
+const Hosting = ( props ) => {
+	const {
+		teams,
+		clickActivate,
+		isECommerceTrial,
+		isMigrationTrial,
+		isHostingTrial,
+		isWpcomStagingSite,
+		siteId,
+		siteSlug,
+		translate,
+		isLoadingSftpData,
+		hasAtomicFeature,
+		hasSftpFeature,
+		hasStagingSitesFeature,
+		isJetpack,
+	} = props;
+
+	const [ transferRefetchEnabled, setTransferRefetchEnabled ] = useState( hasSftpFeature );
+
+	const { isTransferring, transferStatus } = useAtomicTransferQuery( siteSlug, {
+		refetchInterval: 5000,
+		enabled: transferRefetchEnabled,
+	} );
+
+	const isSiteAtomic =
+		transferStatus === transferStates.COMPLETE || transferStatus === transferStates.COMPLETED;
+
+	if ( endStates.includes( transferStatus ) && transferRefetchEnabled ) {
+		// the result won't chnage so no need to call the API
+		setTransferRefetchEnabled( false );
 	}
```

Comment:
Thanks for suggesting this. We can't use `transferStatus` here because it comes from `useAtomicTransferQuery`, which also receives the `transferRefetchEnabled` during initialisation. If I switch to the code below (which I think is what you mean), I get a `ReferenceError: Cannot access 'transferStatus' before initialization`. 

But anyway, I think we should not disable refetching, so we can support the other scenarios you tested, like tabs that were opened before activation. 

```ts
	const { isTransferring, transferStatus } = useAtomicTransferQuery( siteSlug, {
		refetchInterval: 5000,
		enabled: endStates.includes( transferStatus ) && hasSftpFeature,
	} );
```